### ISSUE TITLE
[MaterialSwitch] Thumb stretching

### ISSUE THREAD

**Issue by pubiqq** [material-components/material-components-android#3202]
If you abruptly switch the state of `MaterialSwitch` with a thumb, it will stretch, but only at the ends of the track. I'm sure it shouldn't be like this:

https://user-images.githubusercontent.com/82187521/215922723-f26733ed-6353-429f-bd7e-ac4cea7557bc.mp4

**Material Library version:** 1.8.0

**Comment by drchen** [comment]
Hi by "abruptly switch the state", are you trying to drag the thumb very fast or?

**Comment by pubiqq** [comment]
Yes, almost. I meant it's important not just to drag a thumb quickly, but also to release your finger quickly when reaching the end position.

**Comment by drchen** [comment]
So unfortunately we have very limited control over that - for a finer animation during dragging, we need to tightly couple the animation logic with the dragging & releasing logic, but the later logic is basically implemented inside the SwitchCompat class that is hard for us to "inject" logic there.

This will probably be a low-priority in our task list - but we welcome any ideas that may improve this experience.
